compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -884,14 +884,38 @@ trait Implicits { self: Typer =>
    *
    *     <parent> {
    *       MirroredMonoType = <monoType>
-   *       MirroredTypeConstrictor = <tycon>
-   *       MirroredLabel = <label> }
+   *       MirroredType = <mirroredType>
+   *       MirroredLabel = <label>
+   *       Scope = <scope> // if scope ne NoType
+   *     }
    */
-  private def mirrorCore(parentClass: ClassSymbol, monoType: Type, mirroredType: Type, label: Name)(implicit ctx: Context) = {
-    parentClass.typeRef
+  private def mirrorCore(parentClass: ClassSymbol, monoType: Type, mirroredType: Type, label: Name, extras: List[(Name, Type)])(implicit ctx: Context) = {
+    val m = parentClass.typeRef
       .refinedWith(tpnme.MirroredMonoType, TypeAlias(monoType))
       .refinedWith(tpnme.MirroredType, TypeAlias(mirroredType))
       .refinedWith(tpnme.MirroredLabel, TypeAlias(ConstantType(Constant(label.toString))))
+
+    extras.foldLeft(m) { case (m, (tpnme, tpe)) => m.refinedWith(tpnme, tpe) }
```

Comment:
Would it make sense to just do `formal & m` ?

Reply:
I'll have a look ...

Reply:
Yes, it makes sense, and simplifies things significantly. Good call :-)